distributed/client.py

```diff
@@ -48,6 +49,15 @@
 _global_client = [None]
 
 
+def _timeout_wrapper(timeout, coro_func, *args, **kwargs):
+    coro = coro_func(*args, **kwargs)
+    if timeout is None:
+        return coro
+    else:
+        # XXX should this reraise concurrent.futures.TimeoutError?
+        return gen.with_timeout(timedelta(seconds=timeout), coro)
```

Comment:
I thought about that, but presently `timeout` can be one of the arguments forwarded to sync's `func` callable argument.